### PR TITLE
New create map first modal

### LINKED ISSUE #3388: Pressing enter in the privacy modal just closes the modal
A user would expect that the enter key applies the change, at least for the case of "password-protected" maps where a keyboard input is expected.

In the other cases, a privacy option can be selected but pressing enter doesn't apply them.

image

Not sure if this is an expected behaviour :smile:  cc @viddo @xavijam @saleiva 

Edit: the behaviour happens in dataset privacy modal and map privacy modal.

### PR DESCRIPTION
**Pull request by viddo**
fix-create-map-first

This PR solves the issue that a map needs to be created from current dataset before it can be published or have more layers added. Basically, the new modal replaces the current "create new map first" dialog that already exists, with one using the new styles.

This also fixes #3388 (enter key not saving modals), which was because some internal changes to the base modal that just closed the modal as a default. This fix also removed some boilerplate code for the existing modals.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3429/))

**Comment by viddo** [comment]
@xavijam please code review, cc @saleiva for the visual, looks OK?
deployed on staging ded02 if you guys want to give it a spin

**Comment by saleiva** [comment]
+1

File: lib/assets/javascripts/cartodb/new_common/dialogs/create_vis_first/create_vis_first_view.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * Create a vis from a dataset, required for some contexts to have a vis before be able to carry out next task
+ *  - duplicate vis
+ *  - add layer
+ */
+module.exports = BaseDialog.extend({
+
+  initialize: function() {
+    this.elder('initialize');
+    if (!this.model) throw new Error('model is required (cdb.admin.Visualization)');
+    if (!this.options.title) throw new Error('title is required');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/new_common/dialogs/create_vis_first/create_vis_first_view.js]: We should start thinking about a views validator :)

**Comment by xavijam** [comment]
Looks good :+1: :gb:

File: lib/assets/javascripts/cartodb/new_common/dialogs/create_vis_first/create_vis_first_view.js (hunk 1 above)
Comment (viddo) [inline lib/assets/javascripts/cartodb/new_common/dialogs/create_vis_first/create_vis_first_view.js]: Yep, actually I'm not sure this is the best approach, but until we have something better I think it's better to throw errors early on if prerequisites are not met (and thus gets logged), so we can fix them.

Comment (xavijam) [inline lib/assets/javascripts/cartodb/new_common/dialogs/create_vis_first/create_vis_first_view.js]: agree.

**Comment by viddo** [comment]
cc @iriberri should we add this to the smoke tests doc?

**Comment by iriberri** [comment]
Wow, sure, I'm adding it ;) Thanks!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3438/))

**Pull request merged**